typescript/cli/src/hook/read.ts

```diff
@@ -0,0 +1,54 @@
+import { ChainName, EvmHookReader } from '@hyperlane-xyz/sdk';
+import { Address, ProtocolType } from '@hyperlane-xyz/utils';
+
+import { readChainConfigsIfExists } from '../config/chain.js';
+import { getMultiProvider } from '../context.js';
+import { log, logBlue, logRed } from '../logger.js';
+import { FileFormat, writeFileAtPath } from '../utils/files.js';
+
+/**
+ * Read Hook config for a specified chain and address, logging or writing result to file.
+ */
+export async function readHookConfig({
+  chain,
+  address,
+  chainConfigPath,
+  concurrency = 20,
+  format,
+  output,
+}: {
+  chain: ChainName;
+  address: Address;
+  chainConfigPath: string;
+  concurrency: number;
+  format: FileFormat;
+  output?: string;
+}): Promise<void> {
+  const customChains = readChainConfigsIfExists(chainConfigPath);
+  const multiProvider = getMultiProvider(customChains);
+
+  // output file format overrides provided format
+  format = output?.endsWith('.json')
+    ? 'json'
+    : output?.endsWith('.yaml')
+    ? 'yaml'
+    : format;
```

Comment:
Would the `readYamlOrJson` or related utils in typescript/cli/src/utils/files.ts help with this? If not, consider making a util there for this logic

Reply:
agreed, can we pull this into a shared handler for all of these commands?